### PR TITLE
Feature/5731 delete options

### PR DESCRIPTION
**Pull request by saminahbab**
[Trello Ticket Here ](https://trello.com/c/UQJI7yLK/5731-new-endpoint-in-filter-flex-api-delete-filter-filterid-dimension-dimension-options)
### What

Add interface, methods, route to delete all options of a dimension
Add test to represent all cases 
Add test step to check empty dimension option 

### How to review

run make test and make test-component to ensure no regression 
check steps to ensure all failure cases have been tested 

### Who can review

Team B

### CODE REVIEW

**Review by PauloPortugal — COMMENTED** [review]
Commit subjects provide little insight to what has been done and why 
`pr: address comments` just adds to git log noise without adding what has been addressed and why.
Normally if this is still part of the same work an interactive rebase or a commit amend is the preferred approach.

**Review by ChrysmOre — COMMENTED** [review]
Small comment, also I've noticed when you're naming files you're not being specific enough. Example: options.go, delete-options.feature.. Should be fliter-dimension-options.go and delete-filter-dimension-options.feature. Remember we have both filters and filterOutputs that dimension and options can relate to.
Branch needs rebase

File: api/options.go [hunk 1]
```diff
@@ -72,7 +74,71 @@ func (api *API) getFilterDimensionOptions(w http.ResponseWriter, r *http.Request
...
+					logData: log.Data{
+						"expected_etag": eTag,
+						"actual_etag":   filter.ETag,
+					},
+				},
+			)
+		}
+		return
+	}
+
```
[the remark this answers is not in the document]
Comment (saminahbab) [reply api/options.go:118]: line 126 handles the case that a dimension is not found and returns a 404

[the remark this answers is not in the document]
Comment (saminahbab) [reply api/options.go:118]: > 

Yeah no problem fair comment, I'll change the filenames to be more specific. Worth noting, however,  that many files from the team across the repo are ambiguously named

**Review by PauloPortugal — DISMISSED** [review]

File: api/options.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (saminahbab) [reply api/options.go]: > Small comment, also I've noticed when you're naming files you're not being specific enough. Example: options.go, delete-options.feature.. Should be fliter-dimension-options.go and delete-filter-dimension-options.feature. Remember we have both filters and filterOutputs that dimension and options can relate to. Branch needs rebase

files renamed and noted for future

**Review by PauloPortugal — DISMISSED** [review]

**Review by ChrysmOre — APPROVED** [review]
